### ISSUE TITLE
Bicycle and bus lanes should not be taken into account for lane-indicator

### ISSUE THREAD

**Issue by dktue** [osmandapp/OsmAnd#21402]
### Description

The lane indicator is showing a total of five lanes while probably only showing the two car lanes would be the most helpful for a car driver.
![IMG_20241107_165504](https://github.com/user-attachments/assets/96b54906-6ae2-4309-8c15-38a5abfff1f3)

The situation is [here](https://www.openstreetmap.org/#map=19/48.522204/9.057444), the aerial view shows clearly the lane situation and the position as well as the viewing direction of the photo above. [Google Streetview](https://www.google.de/maps/@48.5223773,9.0571441,3a,75y,119.83h,83.91t/data=!3m7!1e1!3m5!1sk3ojTLgaKDTVL9kQgMb9xQ!2e0!6shttps:%2F%2Fstreetviewpixels-pa.googleapis.com%2Fv1%2Fthumbnail%3Fcb_client%3Dmaps_sv.tactile%26w%3D900%26h%3D600%26pitch%3D6.087952707367776%26panoid%3Dk3ojTLgaKDTVL9kQgMb9xQ%26yaw%3D119.82513953050146!7i16384!8i8192?entry=ttu&g_ep=EgoyMDI0MTExNy4wIKXMDSoASAFQAw%3D%3D) also shows very well how the actual situation looks like.
![image](https://github.com/user-attachments/assets/91e00655-62d5-4fcd-8271-4a2bbf0798b3)

![image](https://github.com/user-attachments/assets/7b5ea023-cbdc-445a-b5e3-a7f3ba932cda)

### Steps to reproduce

-

### Actual result

-

### Expected result

-

### Your Environment (required)

OsmAnd~ 4.9.0#36r, 2024-10-31

**Comment by vmicho** [comment]
The way (https://www.openstreetmap.org/way/531930977) is missing "vehicle:lanes=yes|no|yes|no|yes" (or ...|no as the last lane is BUS). Maybe this is the problem?
However lanes=3 is correct :thinking:

**Comment by dktue** [comment]
I see that the tagging could be better, but: We cannot fix the entire world and I think with some heuristics it might be possible to detect what's actually going on (e.g. process `bicycle:lanes`).

**Comment by vshcherb** [comment]
Though I have at least 2 countries example where it's ok to turn right (not go straight) from the "bus-taxi lane". So could it be country specific?

**Comment by vmicho** [comment]
yeah, access:lanes is more general than vehicle:lanes, so vehicle:lanes should not be needed in this case

**Comment by vshcherb** [comment]
Getting back to the issue, I still couldn't clarify what the algorithm would look like...

```access:lanes = ||||no``` could be used to hide the right lane.

However there are 2 more bivycle lanes to hide.

```bicycle:lanes = | designated || designated |``` - and I'm not sure if designated for bicycle means that's couldn't be used for vehicle. It feels confusing in order to make decision for vehicle lanes we need to go over all types of access and check what means designated and if it's type compatible with vehicle.

**Comment by jendrikw** [comment]
Can you also consider `bus:lanes`?

**Comment by vshcherb** [comment]
To do (Test on https://www.openstreetmap.org/way/531930977):
- [ ] We need to make sure that tags `access:lanes`, `width:lanes`, `bicycle:lanes`, `bus:lanes` available in Turn Lanes widget, so the situation could be drawn correctly. If we could draw an icon of the bus / bicycle based lane availability that would be awesome
- [ ] When we calculate lane availability we need to exclude lanes for driving profile as they non-existing with `[access:lanes]=no`, `[vehicle:lanes]=no`,
- [ ] Awesome support for cycling would be if we follow tag `bicycle:lanes` if it's present.
```
 turn:lanes=left|through|through|right
 vehicle:lanes=yes|yes|no|yes
 bicycle:lanes=yes|no|designated|yes
 cycleway:lanes=no|no|lane|no
```

**Comment by RZR-UA** [comment]
Another place to fix: 48.77287, 9.16745

<img width="972" height="484" alt="Image" src="https://github.com/user-attachments/assets/5e0b7ef9-41fd-4447-b42f-bf9002786bce" />

**Comment by vshcherb** [comment]
Merging #22769 in here — same gap, different junction: the lane indicator counts lanes a car may not use as available. #22769 names the tag that describes it, `access:lanes` (a fourth lane that continues straight for buses only, shown by the widget as a normal through lane).

Whatever filters bicycle lanes out here should read `access:lanes` / `bus:lanes` / `psv:lanes` as well, so one fix covers both reports.

(gen by AI)
